part-customization.tex

```diff
@@ -50,18 +50,77 @@ \section{Introduction}
+if the \texttt{Post} action is executed on a \texttt{saved} invoice, its new state
+will be \texttt{posted}.  E-mailing an invoice does not change its state: it
+remains posted.  Instead, a new workflow is created which manages state of the e-mail.
+
+\digraph[scale=0.4]{wf1}{
+	rankdir=LR;
+	subgraph invoice {
+		graph [label="Invoice"];
+		cluster = true;
+		saved -> posted [label="post"];
+		posted -> reversed [label="reverse"];
+	};
+
+    subgraph email {
+    	graph [label="email"];
+    	cluster = true;
+    	created -> sent [label="send"];
+    };
+
+    posted -> created [label="e-mail"];
+    posted -> posted [label="e-mail"];
+
+}
+
+Actions belonging to a state may (or may not) be available. This is determined by one or
+more conditions.  Examples are ``is the posting date of the transaction in a closed period''
+and ``is the current user allowed to post transactions''.  When the condition evaluates to
+\texttt{false}, the action is not available.
+
+Workflows are configurable. Configurations list the following aspects:
+
+\begin{enumerate}
+	\item A list of states
+	\item A list of allowable actions per state
+	\item A list of conditions per action
+	\item A mapping of actions to Perl code
+	\item A mapping of conditions to Perl code
+\end{enumerate}
+
+With these ingredients, the software's behavior can be changed.  E.g. by adding a new state
+and action in the invoice, an extra reviewing pairs of eyes can be added to the posting process.
+By adding a condition on this action, it can be made required for invoices over 100.000 USD
+(but not for other invoices).
+
+\digraph[scale=0.4]{wf2}{
+	rankdir=LR;
+	subgraph invoice {
+		graph [label="Invoice"];
+		cluster=true;
+		saved -> posted [label="post (<100.000)"];
+		saved -> submitted [label="submit (>100.000)"];
+		submitted -> posted [label="post"];
+		posted -> reversed [label="reverse"]
+	}
+}
+
+By changing association of the \texttt{save} action with its code, the application will act
+differently when saving the invoice.  Customizations may associate new behaviors with existing
+actions or create new ones entirely.  The behavior does not need to be known by the application
+developers, as the association may point to code added to the local installation.  This way, a
+\texttt{remind} action can be added to posted invoices, triggering a text (SMS) message to the
+customer.
 
-Actions associated with a state may be available or not determined by one or more conditions.
-Examples are ``is the posting date of the transaction in a closed period''
-and ``is the current user allowed to post transactions''.
 
 Workflows are stored in the \texttt{./workflows/} directory.  Each workflow consists of
-several files, prefixed with the name of the workflow (e.g. \texttt{ar-ap.}):
+several files, prefixed with the name of the workflow (e.g. \texttt{ar-ap.} for 'AR/AP' workflows):
 
 \begin{enumerate}
 	\item [\texttt{actions.xml}] (optional) names the list of available transitions

```

Comment:
It is confusing to me that the examples above mentioning the workflow name prefix do or do not apply to the xml file names below. I expected each xml file name to have a workflow name prefix or some other explanation.

Reply:
They do need the prefix. What do you propose I do to change your confusion?

Reply:
I would edit the following table to show the prefixes. For example: change each item from

`\item [\texttt{actions.xml}] . . . .`

to

`\item [\texttt{ar-ap.actions.xml}] . . . .`

Reply:
ok. The reason I didn't do that is because the resulting PDF document then looks like this:

![Screenshot from 2024-08-01 18-32-35](https://github.com/user-attachments/assets/f772d5f3-de2b-44e6-8bb5-1e25b740a6d0)

In other words, the list items are too long to even fit the document margin. I'll commit this change for now and merge the PR, but if you have a good idea on the PDF rendering, that'd be very helpful!

Reply:
Yes, I do.  I'll submit a PR after this is merged